src/test/java/core/basesyntax/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,123 @@
+package core.basesyntax;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import core.basesyntax.service.InvalidDataException;
+import core.basesyntax.service.RegistrationServiceImpl;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class RegistrationServiceImplTest {
+    private static RegistrationServiceImpl registrationService;
+    private User testUser;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @BeforeEach
+    void setUp() {
+        testUser = new User();
+    }
+
+    @Test
+    void register_newUser_Ok() {
+        testUser.setLogin("validTestUser");
+        testUser.setAge(50);
+        testUser.setPassword("incred1blePassw0rd");
+        assertEquals(testUser, registrationService.register(testUser));
+    }
+
+    @Test
+    void register_userExists_notOk() {
+        testUser.setLogin("unitTestUserLogin");
+        Storage.people.add(testUser);
+        assertThrows(InvalidDataException.class, () -> registrationService.register(testUser));
+    }
+
+    @Test
+    void register_invalidLogin_notOk() {
+        testUser.setAge(20);
+        testUser.setPassword("password");
+        testUser.setLogin(null);
+        assertThrows(InvalidDataException.class, () -> registrationService.register(testUser));
+        testUser.setLogin("");
+        assertThrows(InvalidDataException.class, () -> registrationService.register(testUser));
+        testUser.setLogin("abc");
+        assertThrows(InvalidDataException.class, () -> registrationService.register(testUser));
+        testUser.setLogin("abcde");
+        assertThrows(InvalidDataException.class, () -> registrationService.register(testUser));
```

Comment:
Consider adding a test case for a login that is exactly the minimum length to ensure boundary conditions are handled correctly.